pkg/providers/snow/reconciler/reconciler_test.go

```diff
@@ -0,0 +1,281 @@
+	machineConfigWN := createSnowWNMachineConfig(true)
+
+	client := fake.NewClientBuilder().WithObjects(cluster, managementCluster, datacenterConfig, bundle, machineConfigCP, machineConfigWN, eksd).Build()
+
+	_, e := reconciler.New(client).Reconcile(context.Background(), log.Log, cluster)
+
+	g.Expect(e).To(BeNil())
+	g.Expect(cluster.Status.FailureMessage).To(BeZero())
+}
+
+func TestSnowClusterReconcileFailDueToInvalidWNMachineConfig(t *testing.T) {
+	g := NewWithT(t)
+
+	managementCluster := createSnowCluster()
+	managementCluster.Name = "management-cluster"
+	cluster := createSnowCluster()
+	cluster.Spec.ManagementCluster = anywherev1.ManagementCluster{Name: "management-cluster"}
+
+	datacenterConfig := createSnowDataCenter(cluster)
+
+	bundle := createBundle(cluster)
+	cluster.Spec.BundlesRef = &anywherev1.BundlesRef{
+		APIVersion: bundle.APIVersion,
+		Name:       bundle.Name,
+		Namespace:  bundle.Namespace,
+	}
+
+	eksd := createEksdRelease()
+	machineConfigCP := createSnowCPMachineConfig(true)
+	machineConfigWN := createSnowWNMachineConfig(false)
+	mcFailureMessage := "Something wrong"
+	machineConfigWN.Status.FailureMessage = &mcFailureMessage
+
+	client := fake.NewClientBuilder().WithObjects(cluster, managementCluster, datacenterConfig, bundle, machineConfigCP, machineConfigWN, eksd).Build()
+
+	_, e := reconciler.New(client).Reconcile(context.Background(), log.Log, cluster)
+	g.Expect(e).To(BeNil(), "error should be nil to prevent requeue")
+
+	g.Expect(cluster.Status.FailureMessage).ToNot(BeZero())
+	g.Expect(*cluster.Status.FailureMessage).To(ContainSubstring("SnowMachineConfig snow-test-wn is invalid"))
+	g.Expect(*cluster.Status.FailureMessage).To(ContainSubstring("Something wrong"))
+}
+
+func TestSnowClusterReconcileFailDueToInvalidCPMachineConfig(t *testing.T) {
+	g := NewWithT(t)
+
+	managementCluster := createSnowCluster()
+	managementCluster.Name = "management-cluster"
+	cluster := createSnowCluster()
+	cluster.Spec.ManagementCluster = anywherev1.ManagementCluster{Name: "management-cluster"}
+
+	datacenterConfig := createSnowDataCenter(cluster)
+
+	bundle := createBundle(cluster)
+	cluster.Spec.BundlesRef = &anywherev1.BundlesRef{
+		APIVersion: bundle.APIVersion,
+		Name:       bundle.Name,
+		Namespace:  bundle.Namespace,
+	}
+
+	eksd := createEksdRelease()
+	machineConfigCP := createSnowCPMachineConfig(false)
+	machineConfigWN := createSnowWNMachineConfig(true)
+
+	client := fake.NewClientBuilder().WithObjects(cluster, managementCluster, datacenterConfig, bundle, machineConfigCP, machineConfigWN, eksd).Build()
+
+	_, e := reconciler.New(client).Reconcile(context.Background(), log.Log, cluster)
+	g.Expect(e).To(BeNil(), "error should be nil to prevent requeue")
+
+	g.Expect(cluster.Status.FailureMessage).ToNot(BeZero())
+	g.Expect(*cluster.Status.FailureMessage).To(ContainSubstring("SnowMachineConfig snow-test-cp is invalid"))
+}
+
+func createSnowCluster() *anywherev1.Cluster {
```

Comment:
nit: for all these method I would suggest to just use the name of the object you are returning, like 
```suggestion
func snowCluster() *anywherev1.Cluster {
```

These methods are not creating any object in the cluster, they just return a struct.